runtime/src/main/kotlin/kotlin/random/Random.kt

```diff
@@ -28,39 +28,12 @@ public abstract class NativeRandom {
             _seed.value = seed
         }
 
-        /**
-         * Returns a pseudo-random Int number.
-         */
-        override fun nextInt(): Int {
-            update((seed * MULTIPLIER + 0xbL) and ((1L shl 48) - 1));
-            return (seed ushr 16).toInt();
+        override fun nextBits(bitCount: Int): Int {
+            update((seed * MULTIPLIER + 0xbL) and ((1L shl 48) - 1))
+            return (seed ushr (48 - bitCount)).toInt()
         }
 
-        /**
-         * Returns a pseudo-random Int value between 0 and specified value (exclusive)
-         */
-        override fun nextInt(bound: Int): Int {
-            if (bound <= 0) throw IllegalArgumentException("Incorrect bound: $bound")
-
-            if (bound and (bound - 1) == 0) {
-                return ((bound * (nextInt() ushr 1).toLong()) ushr 31).toInt();
```

Comment:
I don't think this marginal implementation difference deserves its own reimplementation that has to be maintained and tested.

Reply:
Could you please clarify how it shall affect clients of RNG?

Reply:
Perhaps getting a random number in [0..bound) range where bound is 2^n will be not so fast as in JVM, until we make `fastLog2` (or `Int.numberOfLeadingZeroes` operation) really fast.

Reply:
we have https://github.com/JetBrains/kotlin-native/blob/79755d8fbfaf495e642fc41348b99029c4f6fbde/runtime/src/main/kotlin/kotlin/native/internal/HexStringParser.kt#L272
It looks the same as in JVM so we can make `Int.numberOfLeadingZeroes` be as fast as in the JVM (without taking jit into account)